blocks/header/header.js

```diff
@@ -189,20 +191,69 @@ function settingAltTextForSearchIcon() {
   searchImage.setAttribute('title', listOfAllPlaceholdersData.searchAltText);
 }
 
-async function fetchingPlaceholdersData() {
+async function fetchingPlaceholdersData(placeholdersData) {
   listOfAllPlaceholdersData = await fetchLanguagePlaceholders();
   const hamburger = document.querySelector('.nav-hamburger');
-  hamburger.setAttribute('title', listOfAllPlaceholdersData.hamburgerAltText);
+  hamburger.setAttribute('title', placeholdersData.hamburgerAltText);
   makeImageClickableNSettingAltText();
   settingAltTextForSearchIcon();
 }
-
+const toggleExpandLanguageSelector = (e) => {
+  const toggleContainer = e.currentTarget;
+  if (toggleContainer) {
+    const toggler = toggleContainer.querySelector('div.language-toggle');
+    const toggleContent = toggleContainer.querySelector('div.language-content');
+    if (e.type === 'click' && toggleContent) {
+      e.preventDefault();
+      if (toggleContainer.classList.contains('nav-item-expanded-active')) {
+        toggleContainer.classList.remove('nav-item-expanded-active');
+        toggler.setAttribute('aria-expanded', false);
+        toggleContent.classList.remove('nav-item-content-expanded');
+      } else {
+        toggleContainer.classList.add('nav-item-expanded-active');
+        toggler.setAttribute('aria-expanded', true);
+        toggleContent.classList.add('nav-item-content-expanded');
+      }
+    } else if (e.type === 'mouseenter' && isDesktop.matches && toggleContent) {
+      toggleContainer.classList.add('nav-item-expanded-active');
+      toggler.setAttribute('aria-expanded', true);
+      toggleContent.classList.add('nav-item-content-expanded');
+    } else if (e.type === 'mouseleave' && toggleContent) {
+      toggleContainer.classList.remove('nav-item-expanded-active');
+      toggler.setAttribute('aria-expanded', false);
+      toggleContent.classList.remove('nav-item-content-expanded');
+    }
+  }
+};
+const fetchLanguageSelector = (placeholdersData, alternateMetaLang) => {
+  const ulElement = ul();
+  if (alternateMetaLang) {
+    const metaLangContent = alternateMetaLang.getAttribute('content');
+    if (metaLangContent && metaLangContent.split(',').length > 0) {
+      const langPairs = metaLangContent.split(',');
+      langPairs.forEach((pair) => {
+        const [language, url] = pair.split('|').map((part) => part.trim());
+        const languageDisplayText = placeholdersData[language] || language;
+        ulElement.append(
+          li(
+            a(
+              { href: `${url}` },
+              `${languageDisplayText}`,
+            ),
+          ),
+        );
+      });
+    }
+  }
+  return ulElement;
+};
 export default async function decorate(block) {
   // load nav as fragment
   const navMeta = getMetadata('nav');
-  const lang = getLanguage();
+  const lang = getLanguage() || 'en';
```

Comment:
getLanguage() will return 'en' as default, no need to handle default case.

Reply:
Done